zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -896,3 +901,255 @@ impl<F: Field> MulAddWordsGadget<F> {
         self.overflow.clone()
     }
 }
+
+#[derive(Clone, Debug)]
+pub(crate) struct ShrWordsGadget<F> {
+    a: util::Word<F>,
+    shift: util::Word<F>,
+    b: util::Word<F>,
+    // four 64-bit limbs of word `a`
+    a64s: [Cell<F>; 4],
+    // four 64-bit limbs of word `b`
+    b64s: [Cell<F>; 4],
+    // lower `n` bits
+    a64s_lo: [Cell<F>; 4],
```

Comment:
Perhaps a brief explanation of `a64s_lo` and `a64s_hi`? Something like: Each of the four `a64s` limbs is split into two parts (`lo` and `hi`) at position `shf_mod64`.